api/test/CMakeLists.txt

```diff
@@ -13,3 +13,4 @@ if(WITH_LOGS_PREVIEW)
 endif()
 add_subdirectory(common)
 add_subdirectory(baggage)
+add_subdirectory(singleton)
```

Comment:
Should we add this subdirectory only for the non-windows platform? This will ensure we don't build the singleton tests for both bazel and cmake.

Reply:
Please ignore. My bad, this file is only for cmake. I think it is fine to build the bazel tests for windows if we can't avoid that.